File: 914-x-of-a-kind-in-a-deck-of-cards/914-x-of-a-kind-in-a-deck-of-cards.cpp
class Solution {
private:
    int gcd(int a, int b) {
        return b > 0 ? gcd(b, a % b) : a;
    }
public:
    bool hasGroupsSizeX(vector<int>& deck) {
        int ans = 0;
        unordered_map<int, int> mp;
        for(int card: deck) mp[card]++;
        for(auto freq : mp) ans = gcd(freq.second, ans);
        return ans > 1;
    }
};

File: 914-x-of-a-kind-in-a-deck-of-cards/914-x-of-a-kind-in-a-deck-of-cards.java
class Solution {
    private int gcd(int a, int b) {
        return b > 0 ? gcd(b, a % b) : a;
    }
    public boolean hasGroupsSizeX(int[] deck) {
        int ans = 0;
        var map = new HashMap<Integer, Integer>();
        for(int card : deck) map.put(card, map.getOrDefault(card, 0) + 1);
        for(int freq : map.values()) ans = gcd(freq, ans);
        return ans > 1;
    }
}